ksqldb-functional-tests/src/main/java/io/confluent/ksql/test/tools/TestExecutor.java

```diff
@@ -499,53 +500,52 @@ private void processSingleRecord(
       final Set<Topic> possibleSinkTopics
   ) {
     final String topicName = producedRecord.topic();
-
-    final ConsumerRecord<byte[], byte[]> consumerRecord =
-        new org.apache.kafka.streams.test.ConsumerRecordFactory<>(
-            new ByteArraySerializer(),
-            new ByteArraySerializer()
-        ).create(
-            topicName,
-            producedRecord.key(),
-            producedRecord.value(),
-            producedRecord.timestamp()
-        );
-
-    testDriver.getTopologyTestDriver().pipeInput(consumerRecord);
+    final TestInputTopic<byte[], byte[]> inputTopic = testDriver
+        .getTopologyTestDriver()
+        .createInputTopic(topicName, new ByteArraySerializer(), new ByteArraySerializer());
```

Comment:
It seems actually incorrect, to create a new `TestInputTopic` over an over again -- we should instead create it only once.